core/interfaces/i_toolbox.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a toolbox.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IToolbox');
+
+goog.require('Blockly.IUIPlugin');
+
+/**
+ * @extends {Blockly.IUIPlugin}
+ * @interface
+ */
+Blockly.IToolbox = function() {};
+
+/**
+ * Get the toolbox flyout.
+ * @return {Blockly.Flyout} The toolbox flyout.
+ */
+Blockly.IToolbox.prototype.getFlyout;
+
+/**
+ * Unhighlight any previously specified option.
+ * @return {void}
+ */
+Blockly.IToolbox.prototype.clearSelection;
+
+/**
+ * Updates the category colours and background colour of selected categories.
+ */
+Blockly.IToolbox.prototype.updateColourFromTheme;
+
+/**
+ * Get the width of the toolbox.
+ * @return {number} The width of the toolbox.
+ */
+Blockly.IToolbox.prototype.getWidth;
+
+/**
+ * Get the height of the toolbox.
+ * @return {number} The width of the toolbox.
+ */
+Blockly.IToolbox.prototype.getHeight;
+
+/**
+ * Return the deletion rectangle for this toolbox.
+ * @return {Blockly.utils.Rect} Rectangle in which to delete.
+ * TODO: Should be part of deletable?
```

Comment:
This is used for delete areas rather than deletable objects which is what IDeletable is intended for. I would say an `IDeleteArea` interface that Toolbox / Flyout / Trashcan can implement.